sktime/forecasting/compose/_fallback.py

```diff
@@ -0,0 +1,139 @@
+    ...         EnsembleForecaster(
+    ...             [
+    ...                 ("trend", PolynomialTrendForecaster()),
+    ...                 ("polynomial", NaiveForecaster())
+    ...             ]
+    ...         )
+    ...     ),
+    ...     ("naive", NaiveForecaster())
+    ... ]
+    >>> forecaster = FallbackForecaster(forecasters=forecasters)
+    >>> forecaster.fit(y=y, fh=[1, 2, 3])
+    FallbackForecaster(...)
+    >>> y_pred = forecaster.predict()
+
+    """
+
+    def __init__(self, forecasters, warn=False):
+        super(FallbackForecaster, self).__init__()
+        self.forecasters = forecasters
+        self.current_forecaster = None
+        self.current_name = None
+        self.current_forecaster_index = 0
+        self.warn = warn
+        self._anytagis_then_set("requires-fh-in-fit", True, False, self.forecasters)
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the forecasters in the given order until one succeeds.
+
+        Parameters
+        ----------
+        y : array-like
+            Target time series to which to fit the forecasters.
+        X : array-like, optional (default=None)
+            Exogenous variables.
+        fh : array-like, optional (default=None)
+            The forecasting horizon.
+
+        Returns
+        -------
+        self : an instance of self
+
+        Raises
+        ------
+        RuntimeError
+            If all forecasters fail to fit."""
+        self.y = y
+        for name, forecaster in self.forecasters[self.current_forecaster_index :]:
+            try:
+                forecaster.fit(y, X=X, fh=fh)
+                self.current_forecaster = forecaster
+                self.current_name = name
+                return self
+            except Exception as e:
+                self.current_forecaster_index += 1
+                if self.warn:
+                    warnings.warn(f"Forecaster {name} failed to fit with error: {e}")
+
+        raise RuntimeError("All forecasters failed to fit.")
+
+    def _predict(self, fh, X=None):
+        """Predict using the current forecaster. If it fails, fit and predict with the next forecaster.
+
+        Parameters
+        ----------
+        fh : array-like
+            The forecasting horizon.
+        X : array-like, optional (default=None)
+            Exogenous variables.
+
+        Returns
+        -------
+        y_pred : array-like
+            The predicted values.
+
+        Raises
+        ------
+        RuntimeError
+            If no forecaster is fitted or all forecasters fail to predict.
+        """
+        if self.current_forecaster is None:
+            raise RuntimeError("No forecaster has been successfully fitted yet.")
+
+        try:
+            return self.current_forecaster.predict(fh, X)
+        except Exception as e:
+            warnings.warn(f"Current forecaster failed at prediction with error: {e}")
+
+            # Remove the failed forecaster
+            self.forecasters.pop(0)
```

Comment:
this should not be done - `__init__` args must be written to self and never mutated. Further, `predict` should not have any side effects on anything.